lib/turn.rb

```diff
@@ -0,0 +1,33 @@
+class Turn
+
+  attr_reader :action
+
+  def initialize
+    print "which row?\n > "
+    @row = gets.chomp.to_i
+    print "which column?\n > "
+    @column = gets.chomp.to_i
+    @action = "x"
+    until action_is_valid?
+      print "which action?\n > "
+      @action = gets.chomp.upcase
+      print "that is not a valid action" unless action_is_valid?
+    end
+  end
+
+  def message
+    { row: @row,
+      column: @column,
+      action: @action }
+  end
+
+  private
+
+  def action_is_valid?
+    valid_actions.include? action
```

Comment:
Although it can be annoying to chase up, validation is always key; I'm trying to be better behaved with mine.